### PR TITLE
Contrib guide improv

### PR DESCRIPTION
**Pull request by igncampa**
## Description
clarified a few things in the development docs and added a section to troubleshoot running the tests

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [ ] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by igncampa** [comment]
hmm, any idea what's going on here? I only touched the `DEVELOP.rst` file. I don't see how that can affect the tests in any way

**Comment by j-bennet** [comment]
@igncampa We have a test or two that give intermittent failures. One day I (or someone) hopefully fixes that. I restarted the travis job for now.

File: DEVELOP.rst [hunk 1]
```diff
@@ -147,9 +153,26 @@ To see stdout/stderr, use the following command:
...
     $ behave --no-capture
 
+Troubleshooting the integration tests
+-------------------------------------
+
+If you encounter errors when running the tests you might have to edit your
+``pg_hba.conf`` file. The database user has to have permissions to create and
+drop test databases. The default user is ``postgres`` at ``localhost``. Make
+sure the method is set to ``trust``. You might need to set all the ``local``
+connections to ``trust`` as well.
```
Comment (j-bennet) [inline DEVELOP.rst]: I think this problem is a bug that needs to be fixed. Tests should be able to run on `local` or `host` connection. I would rather not add this part to docs, but file an issue to fix the problem.

File: DEVELOP.rst [hunk 2]
```diff
@@ -147,9 +153,26 @@ To see stdout/stderr, use the following command:
 
     $ behave --no-capture
 
+Troubleshooting the integration tests
+-------------------------------------
+
```
Comment (j-bennet) [inline DEVELOP.rst]: I think this paragraph belongs before the part about running behave tests. We need to make it clear tha it's a prerequisite to running tests. Something like this:

First, install the requirements... etc.
Then, make sure you have default database user `postgres` configured... etc.
Then:
`cd tests`
`behave`

Comment (igncampa) [reply DEVELOP.rst]: Would a pointer to the troubleshooting section do the work? Maybe like:

    First, install the requirements... etc.

    Make sure you have default database user postgres configured.
    **See the troubleshooting section for more information**

    Then:
        cd tests
        behave

Comment (j-bennet) [reply DEVELOP.rst]: Troubleshooting has its place, perhaps, but the normal workflow should come first and take precedence. Yes, our tests are broken in a few places right now, but I don't want the documentation to "normalize" it. Tests being broken is not normal. You should not have trouble running them. I would describe how postgres user should be configured in "normal workflow" section. Then in "troubleshooting" section, you can add a few more pointers, like:
* make sure postgres instance on localhost is running
* check pg_hba config to verify local connections are enabled
* contact us on gitter or file an issue.

**Comment by igncampa** [comment]
Apologies for the amount of commits. I hope we find a common ground with these changes. Let me know otherwise!

File: DEVELOP.rst [hunk 3]
```diff
@@ -123,22 +127,35 @@ in the ``tests`` directory.  An example::
...
+at ``localhost``. Make sure the authentication method is set to ``trust``. If
+you made any changes to your ``pg_hba.conf`` make sure to restart the postgres
+service for the changes to take effect.
+
+::
+
+    # ONLY IF YOU MADE CHANGES TO YOUR pg_hba.conf FILE
+    $ sudo service postgresql restart
+
+After that, tests can be run on the ``/pgcli/tests`` directory with:
```
Comment (j-bennet) [inline DEVELOP.rst]: Should be **in directory**, not **on directory**, right?

**Comment by j-bennet** [comment]
@igncampa This PR is mostly good to go, except for one typo. Please rebase on top of master if possible. Thank you!

**Comment by igncampa** [comment]
Sorry for being away for a while. I admit my git-fu is pretty weak and I'm hesitant to doing anything beyond the regular branching, commiting, pushing/pulling workflow. So I just did:

- `git rebase master`

- fixed the typo. `git commit`

- `git push -f ` as suggested [here](https://stackoverflow.com/questions/39399804/updates-were-rejected-because-the-tip-of-your-current-branch-is-behind#39400690)

hope I didn't screw anything up!

**Comment by j-bennet** [comment]
Nothing screwed up that I can see. :)

Please add your contribution to the changelog and update AUTHORS file!

**Comment by igncampa** [comment]
I'd rather add myself to the AUTHORS file only after having contributed code (soon!), if that's ok with you

**Comment by j-bennet** [comment]
Merged. Thank you for the PR!

🍫

**Pull request merged**